Sources/PrimerSDK/Classes/User Interface/Apple Pay/ApplePayViewModel.swift

```diff
@@ -1,69 +1,317 @@
+    var orderItems: [OrderItem] {
         let settings: PrimerSettingsProtocol = DependencyContainer.resolve()
-        return settings.countryCode
+        return settings.orderItems
+    }
+    var applePayConfigId: String? {
+        let state: AppStateProtocol = DependencyContainer.resolve()
+        return state.paymentMethodConfig?.getConfigId(for: .applePay)
     }
-    var uxMode: UXMode { return Primer.shared.flow.internalSessionFlow.uxMode }
     var clientToken: DecodedClientToken? {
         let state: AppStateProtocol = DependencyContainer.resolve()
         return state.decodedClientToken
     }
+    var isVaulted: Bool {
+        return Primer.shared.flow.internalSessionFlow.vaulted
+    }
+    var uxMode: UXMode {
+        return Primer.shared.flow.internalSessionFlow.uxMode
+    }
+    
+    private var applePayCompletion: ((Result<ApplePayPaymentResponse, Error>) -> Void)?
 
     deinit {
         log(logLevel: .debug, message: "🧨 deinit: \(self) \(Unmanaged.passUnretained(self).toOpaque())")
     }
 
-    func tokenize(instrument: PaymentInstrument, completion: @escaping (Error?) -> Void) {
-        let state: AppStateProtocol = DependencyContainer.resolve()
-        let request = PaymentMethodTokenizationRequest(paymentInstrument: instrument, state: state)
+    // swiftlint:disable cyclomatic_complexity function_body_length
+    func payWithApple(completion: @escaping (Error?) -> Void) {
+        guard let countryCode = countryCode else {
+            return completion(PaymentException.missingCountryCode)
+        }
         
-        let settings: PrimerSettingsProtocol = DependencyContainer.resolve()
-        let tokenizationService: TokenizationServiceProtocol = DependencyContainer.resolve()
-        tokenizationService.tokenize(request: request) { [weak self] result in
-            switch result {
-            case .failure(let error):
-                ErrorHandler.shared.handle(error: error)
-                completion(error)
-            case .success(let token):
-                // This is Apple Pay, it always gonna be checkout
-                settings.authorizePayment(token, completion)
+        guard let currency = currency else {
+            return completion(PaymentException.missingCurrency)
+        }
+        
+        guard let merchantIdentifier = merchantIdentifier else {
+            return completion(AppleException.missingMerchantIdentifier)
+        }
+        
+        guard !orderItems.isEmpty else {
+            return completion(PaymentException.missingOrderItems)
+        }
+        
+        let applePayRequest = ApplePayRequest(
+            currency: currency,
+            merchantIdentifier: merchantIdentifier,
+            countryCode: countryCode,
+//            supportedNetworks: supportedNetworks,
+            items: orderItems
+//            merchantCapabilities: merchantCapabilities
+        )
+        
+        var supportedNetworks: [PKPaymentNetwork] = [
```

Comment:
[Suggestion]
Would be great to have this whole logic of supported networks on another file. It's making reading the code a bit tedious 👍

Reply:
Makes sense. Done!